softgroup/basis/common-router/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java

```diff
@@ -2,17 +2,37 @@
 
 import com.softgroup.common.protocol.Request;
 import com.softgroup.common.protocol.Response;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 public abstract class AbstractRouterHandler<T extends Handler> implements RouterHandler {
 
-	@Override
-	public String getRouteKey(Request<?> msg) {
+	private Map<String, T> handlerMap = new HashMap();
+
+	@Autowired
+	private List<T> handlers;
+
+	@PostConstruct
+	public void init(){
+		for(T handler : handlers)
+			handlerMap.put(handler.getName(), handler);
+	}
+
+	public String getName() {
 		return null;
 	}
 
-	@Override
 	public Response<?> handle(Request<?> msg) {
-		return null;
+		//String command = msg.getHeader().getCommand();
+		return handlerMap.get(getRouteKey(msg)).handle(msg);
```

Comment:
@Dominys So firstly I need to check if needed handler exists, if not - return null, correct?